### PR TITLE
feat: utils and webpack plugin for an index.html entrypoint

### PR DESCRIPTION
**Pull request by LarsDenBakker**
This is part of a larger refactoring. I'm extracting all common code between different build related projects so that they can be reused. For example `building-rollup`, `building-webpack` and `owc-dev-server` all need to extract modules from a `index.html`. 

This also separate the `modern web plugin` as `webpack-index-html-plugin`, so that it can be used separately by power users (I actually want to reuse it separately myself for my project at work).

For more information you can check the readme of `webpack-index-html-plugin`

In this PR I am only updating `building-webpack`, later on I will update the other projects.

Fixes https://github.com/open-wc/open-wc/issues/435

Fixes https://github.com/open-wc/open-wc/issues/475

Fixes https://github.com/open-wc/open-wc/issues/435

### CODE REVIEW

**Comment by LarsDenBakker** [comment]
This is ready for review

File: packages/webpack-index-html-plugin/webpack-index-html-plugin.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      }
+
+      if (config.multiIndex.variations.some(v => !v.name)) {
+        throw createError(`Some variations are missing a name property.`);
+      }
+    }
+  }
+
+  apply(compiler) {
+    // @ts-ignore
```
Comment (daKmoR) [inline packages/webpack-index-html-plugin/webpack-index-html-plugin.js]: added this ignore... is this if needed? (it is nowhere set... does it comes from webpack directly?)

**Comment by daKmoR** [comment]
made some changes and a force push 😬 
- rebased
- made it 2 commits
- use babel 7.3.3 version everywhere
- add some eslint ignore for demos
- set version of the webpack plugin to 0.0.1 (will be bumped to 0.1.0 when released)
- added one comment for a ts-ignore (but that is not blocking)

very nice work 💪 

so LGTM => feel free to merge 🤗 

PS: it seems like something broke vuepress :/ it can't find some of the core-js things... let's fix that in a separate MR... so homepage will not update until that is fixed 🙈

**Comment by daKmoR** [comment]
hmmm scratch that remark for `vuepress`... it works fine on master 😱  
sooo looking at the changes I am all about - "whaatt the hell could break it?" 🤔

File: packages/building-utils/package.json [hunk 2]
```diff
@@ -8,23 +8,35 @@
...
   "files": [
     "src",
+    "index-html",
+    "dom5-fork",
     "*.js"
   ],
   "dependencies": {
-    "@babel/core": "^7.4.4",
+    "@babel/core": "^7.3.3",
+    "@types/deepmerge": "^2.2.0",
```
Comment (daKmoR) [inline packages/building-utils/package.json]: > This is a stub types definition. deepmerge provides its own type definitions, so you do not need this installed.
?

Comment (daKmoR) [reply packages/building-utils/package.json]: I removed it

**Comment by daKmoR** [comment]
ok, I restored the `yarn.lock` file from master and then did an install which now only add the new dependencies...
=> doing that restored vuepress
=> and also the [preview](https://deploy-preview-474--open-wc-org.netlify.com/) is back 🤗 

so the lock currently saves our ass 🙈  at some point we will need to update everything and make sure it still works... but not today 😬

File: packages/webpack-index-html-plugin/webpack-index-html-plugin.js (hunk 1 above)
Comment (LarsDenBakker) [reply packages/webpack-index-html-plugin/webpack-index-html-plugin.js]: Yes apply comes from webpack

**Pull request merged**